IHP/QueryBuilder.hs

```diff
@@ -512,4 +524,31 @@ instance (model ~ GetModelById (Id' table), value ~ Id' table, HasField "id" mod
     {-# INLINE fetchOneOrNothing #-}
     fetchOneOrNothing = genericfetchIdsOneOrNothing
     {-# INLINE fetchOne #-}
-    fetchOne = genericFetchIdsOne
\ No newline at end of file
+    fetchOne = genericFetchIdsOne
+
+-- | Adds an @DISTINCT to your query.
+--
+-- Use 'distinct' to remove all duplicate rows from the result
+--
+-- __Example:__ Fetch distinct books
+--
+-- > query @Book
+-- >     |> distinct
+-- >     |> fetch
+-- > -- SELECT DISTINCT * FROM books
+distinct :: QueryBuilder table -> QueryBuilder table
+distinct = DistinctQueryBuilder
```

Comment:
```suggestion
distinct = DistinctQueryBuilder
{-# INLINE distinct #-}
```

Reply:
There's a lot of explicit inlining in the IHP codebase. How much effect does this actually have in this day & age?

Reply:
We'd need to benchmark this at some point (but that's low prio). In general I guess it helps to specialize type heavy code (like query builder) and avoids passing too many class dicts around.